dev-requirements.txt

```diff
@@ -145,8 +137,6 @@ webencodings==0.5.1
     # via bleach
 websocket-client==0.57.0
     # via docker
-zipp==3.4.0
```

Comment:
I these changes are not up to date with dev-requirements.in since if I pip-compile I do get these 2 dependencies back

Reply:
That's my local output of the command. Not sure what's happening here.

```
❯ pip-compile dev-requirements.in
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile dev-requirements.in
#
appdirs==1.4.4
    # via virtualenv
attrs==20.3.0
    # via pytest
bleach==3.3.0
    # via readme-renderer
certifi==2020.12.5
    # via
    #   -c requirements.txt
    #   requests
chardet==3.0.4
    # via
    #   -c requirements.txt
    #   requests
click==7.1.2
    # via
    #   -c requirements.txt
    #   pip-tools
colorama==0.4.4
    # via twine
coverage==5.4
    # via pytest-cov
distlib==0.3.1
    # via virtualenv
docker==4.4.1
    # via tox-docker
docutils==0.16
    # via readme-renderer
faker==5.3.0
    # via -r dev-requirements.in
filelock==3.0.12
    # via
    #   tox
    #   virtualenv
idna==2.10
    # via
    #   -c requirements.txt
    #   requests
iniconfig==1.1.1
    # via pytest
keyring==22.0.1
    # via twine
more-itertools==8.6.0
    # via pytest
packaging==20.9
    # via
    #   bleach
    #   pytest
    #   tox
pip-tools==5.5.0
    # via -r dev-requirements.in
pkginfo==1.7.0
    # via twine
pluggy==0.13.1
    # via
    #   pytest
    #   tox
py==1.10.0
    # via
    #   pytest
    #   tox
pygments==2.7.4
    # via readme-renderer
pyparsing==2.4.7
    # via packaging
pytest-cov==2.10.1
    # via -r dev-requirements.in
pytest-html==3.1.1
    # via -r dev-requirements.in
pytest-metadata==1.11.0
    # via pytest-html
pytest==6.0.2
    # via
    #   -r dev-requirements.in
    #   pytest-cov
    #   pytest-html
    #   pytest-metadata
python-dateutil==2.8.1
    # via
    #   -c requirements.txt
    #   faker
python-dotenv==0.13.0
    # via -r dev-requirements.in
readme-renderer==28.0
    # via twine
requests-toolbelt==0.9.1
    # via twine
requests==2.23.0
    # via
    #   -c requirements.txt
    #   docker
    #   requests-toolbelt
    #   twine
rfc3986==1.4.0
    # via twine
six==1.15.0
    # via
    #   -c requirements.txt
    #   bleach
    #   docker
    #   python-dateutil
    #   readme-renderer
    #   tox
    #   virtualenv
    #   websocket-client
tabulate==0.8.7
    # via -r dev-requirements.in
text-unidecode==1.3
    # via faker
toml==0.10.2
    # via
    #   pytest
    #   tox
tox-docker==2.0.0
    # via -r dev-requirements.in
tox==3.15.2
    # via
    #   -r dev-requirements.in
    #   tox-docker
tqdm==4.56.0
    # via twine
twine==3.3.0
    # via -r dev-requirements.in
urllib3==1.25.10
    # via
    #   -c requirements.txt
    #   requests
virtualenv==20.4.2
    # via tox
webencodings==0.5.1
    # via bleach
websocket-client==0.57.0
    # via docker

# The following packages are considered to be unsafe in a requirements file:
# pip
# setuptools
```

Reply:
if you execute `scripts/recreate_venv.sh` which removes and rebuilds from the beginning do you still get the same result, without zipp and importlib-metadata?

Can you also say which exact version of all of your dependencies in your venv are via `pip list`?

Reply:
Yes.

```
❯ pip list
Package                    Version   Location
-------------------------- --------- ---------------------------------------------
appdirs                    1.4.4
asn1crypto                 1.4.0
attrs                      20.3.0
azure-common               1.1.25
azure-core                 1.8.2
azure-storage-blob         12.5.0
bleach                     3.3.0
boto3                      1.15.18
botocore                   1.18.18
cachetools                 4.2.0
certifi                    2020.12.5
cffi                       1.14.3
chardet                    3.0.4
click                      7.1.2
colorama                   0.4.4
coverage                   5.4
cryptography               3.3.2
distlib                    0.3.1
docker                     4.4.1
docutils                   0.16
Faker                      5.3.0
filelock                   3.0.12
future                     0.18.2
google-api-core            1.24.1
google-auth                1.24.0
google-cloud-bigquery      2.5.0
google-cloud-core          1.5.0
google-crc32c              1.1.0
google-resumable-media     1.2.0
googleapis-common-protos   1.52.0
grpcio                     1.34.0
idna                       2.10
iniconfig                  1.1.1
isodate                    0.6.0
Jinja2                     2.11.2
jmespath                   0.10.0
keyring                    22.0.1
MarkupSafe                 1.1.1
more-itertools             8.6.0
msrest                     0.6.19
oauthlib                   3.1.0
oscrypto                   1.2.1
packaging                  20.9
pip                        21.0.1
pip-tools                  5.5.0
pkginfo                    1.7.0
pluggy                     0.13.1
proto-plus                 1.13.0
protobuf                   3.14.0
psycopg2-binary            2.8.5
py                         1.10.0
pyasn1                     0.4.8
pyasn1-modules             0.2.8
PyAthena                   1.10.7
pycparser                  2.20
pycryptodomex              3.9.8
Pygments                   2.7.4
PyJWT                      1.7.1
pyodbc                     4.0.30
pyOpenSSL                  19.1.0
pyparsing                  2.4.7
pytest                     6.0.2
pytest-cov                 2.10.1
pytest-html                3.1.1
pytest-metadata            1.11.0
python-dateutil            2.8.1
python-dotenv              0.13.0
pytz                       2020.1
PyYAML                     5.3.1
readme-renderer            28.0
requests                   2.23.0
requests-oauthlib          1.3.0
requests-toolbelt          0.9.1
rfc3986                    1.4.0
rsa                        4.6
s3transfer                 0.3.3
setuptools                 41.2.0
six                        1.15.0
snowflake-connector-python 2.3.6
soda-sql                   0.0.0     /Users/antoninjousson/Documents/code/soda-sql
tabulate                   0.8.7
tenacity                   6.2.0
text-unidecode             1.3
toml                       0.10.2
tox                        3.15.2
tox-docker                 2.0.0
tqdm                       4.56.0
twine                      3.3.0
urllib3                    1.25.10
virtualenv                 20.4.2
webencodings               0.5.1
websocket-client           0.57.0
```

Reply:
I see your and mine `setuptools` are differing (I have 47.1.0 and you have 41.2.0). Maybe that's causing issues and it is not part of our `{dev-,}requirements.txt` files. I have my from python 3.8 installation, and native Mac python installation is 3.7 (and comes with, indeed, 41.2.0 setuptools package). But even then - if I downgrade my setuptools I still get different pip-compile output.

Let's just forget about it and investigate later. It's weird and will cause issues but let's not lose our head on this for now

Reply:
solved by letting @milanaleksic move to Python 3.8